### PR TITLE
Import uptime table and feature from TC

### PR DESCRIPTION
**Pull request by neltharionorg**
Uptime table https://github.com/azerothcore/azerothcore-wotlk/issues/236

### CODE REVIEW

File: src/game/World/World.cpp [hunk 1]
```diff
@@ -1760,11 +1766,16 @@ void World::SetInitialWorldSettings()
...
+    LoginDatabase.PExecute("INSERT INTO uptime (realmid, starttime, uptime, revision) VALUES(%u, %u, 0, '%s')",
+        realmID, uint32(m_startTime), _FULLVERSION);       // One-time query
+
 
 
     m_timers[WUPDATE_WEATHERS].SetInterval(1*IN_MILLISECONDS);
     m_timers[WUPDATE_AUCTIONS].SetInterval(MINUTE*IN_MILLISECONDS);
     m_timers[WUPDATE_AUCTIONS].SetCurrent(MINUTE*IN_MILLISECONDS);
+    m_timers[WUPDATE_UPTIME].SetInterval(m_int_configs[CONFIG_UPTIME_UPDATE] * MINUTE*IN_MILLISECONDS);
+                                                             //Update "uptime" table based on configuration entry in minutes.
```
Comment (ShinDarth) [inline src/game/World/World.cpp]: please fix indentation

File: src/game/World/World.cpp [hunk 2]
```diff
@@ -2085,7 +2096,25 @@ void World::Update(uint32 diff)
...
     ProcessQueryCallbacks();
-
+    
+        	/// <li> Update uptime table
+        if (m_timers[WUPDATE_UPTIME].Passed())
+         {
+        uint32 tmpDiff = uint32(m_gameTime - m_startTime);
+        uint32 maxOnlinePlayers = GetMaxPlayerCount();
+        
+            m_timers[WUPDATE_UPTIME].Reset();
```
Comment (ShinDarth) [inline src/game/World/World.cpp]: please fix indentation

File: src/game/World/World.cpp [hunk 3]
```diff
@@ -2085,7 +2096,25 @@ void World::Update(uint32 diff)
...
+        
+            PreparedStatement* stmt = LoginDatabase.GetPreparedStatement(LOGIN_UPD_UPTIME_PLAYERS);
```
Comment (ShinDarth) [inline src/game/World/World.cpp]: please fix indentation

File: src/game/World/World.cpp [hunk 4]
```diff
@@ -2085,7 +2096,25 @@ void World::Update(uint32 diff)
...
+        
+        stmt->setUInt32(0, tmpDiff);
+        stmt->setUInt16(1, uint16(maxOnlinePlayers));
+        stmt->setUInt32(2, realmID);
+        stmt->setUInt32(3, uint32(m_startTime));
+        
+            LoginDatabase.Execute(stmt);
```
Comment (ShinDarth) [inline src/game/World/World.cpp]: please fix indentation

**Comment by neltharionorg** [comment]
@ShinDarth i dont know how to do that i did something and if its not correct can you tell me how to fix indentation?

**Comment by BarbzYHOOL** [comment]
Shouldn't this be an optional feature when compiling though ? @Yehonal

**Comment by Yehonal** [comment]
Nope it's a foundamental utility for server monitoring

Il 21 Nov 2017 16:03, "Barbz" <notifications@github.com> ha scritto:

**Comment by ShinDarth** [comment]
please, can anyone test this? so we can merge it

**Comment by BarbzYHOOL** [comment]
If people wrote a guide/pattern to test the PR, it would be very useful to test quickly. Maybe update the github PR form default text like you did with the issue form?

**Comment by neltharionorg** [comment]
@BarbzYHOOL compile core with this pr add sql and start server than check uptime table if you have website with uptime feature in server status you can check there too.

**Comment by Yehonal** [comment]
@BarbzYHOOL 

image

**Comment by Yehonal** [comment]
tested on staging. It works ;)

**Pull request merged**